File: ruby/hamming/hamming.rb
class Hamming

  def self.compute(first, second)
    first_array = first.split('')
    second_array = second.split('')
    hamming = 0
    if first_array.count <= second_array.count
      first_array.count.times do |num|
        if first_array[num-1] != second_array[num-1] && !first_array[num-1].empty? && !second_array[num-1].empty?
          print "first " + first_array[num-1] + " second "+ second_array[num-1]
          hamming += 1
        end
      end
    end
    hamming
  end
end

File: javascript/hamming/hamming.js
function compute(strand1, strand2){
  strand1 = strand1.split('');
  strand2 = strand2.split('');
  var len = strand1.length > strand2.length ? strand2.length : strand1.length;
  var counter = 0;
  for (var i=0; i<len; i++){
    if (strand1[i] !== strand2[i]){
      counter++;
    }
  }
  return counter;
}

module.exports = {compute: compute};